src/Endatix.Extensions.Hosting/EndatixWebAppBuilderExtensions.cs

```diff
@@ -0,0 +1,89 @@
+using Ardalis.GuardClauses;
+using Microsoft.AspNetCore.Builder;
+using Logging = Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Extensions.Logging;
+using Endatix.Framework.Hosting;
+using Endatix.Extensions.Hosting;
+using System.Runtime.Versioning;
+using Endatix.Core.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Endatix.Setup;
+
+/// <summary>
+/// Provides extension methods for configuring and setting up the Endatix application within a <see cref="WebApplicationBuilder"/> context as well as  <see cref="IEndatixApp"/> context.
+/// </summary>
+public static class EndatixHostBuilderExtensions
+{
+    /// <summary>
+    /// Creates and configures an instance of <see cref="IEndatixApp"/> using the provided <see cref="WebApplicationBuilder"/>. This method also configures logging using the provided logger or a default Serilog configuration if none is provided.
+    /// </summary>
+    /// <param name="builder">The <see cref="WebApplicationBuilder"/> used to configure the application.</param>
+    /// <param name="logger">Optional. The <see cref="Logging.ILogger"/> instance for logging. If null, a default Serilog logger is used.</param>
+    /// <returns>An instance of <see cref="IEndatixApp"/> representing the configured application.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if the <paramref name="builder"/> is null.</exception>
+    public static IEndatixApp CreateEndatix(this WebApplicationBuilder builder, Logging.ILogger? logger = null)
+    {
+        Guard.Against.Null(builder);
+
+        if (logger == null)

```

Comment:
Similar to comments in a previous PR, I think this is a good candidate for method extraction.

Reply:
@dspetrov  - this was outdated. I have merged latest from `main` and you can see this file has not been changed. Sorry for that.